src/vscode-azurefunctions.api.d.ts

```diff
@@ -9,12 +9,24 @@ export interface AzureFunctionsExtensionApi {
     revealTreeItem(resourceId: string): Promise<void>;
 
     createFunction(options: ICreateFunctionOptions): Promise<void>;
+    downloadAppSettings(client: ISimpleAppSettingsClient): Promise<void>;
+    uploadAppSettings(client: ISimpleAppSettingsClient, workspacePath?: string): Promise<void>;
 }
 
 export type ProjectLanguage = 'JavaScript' | 'TypeScript' | 'C#' | 'Python' | 'PowerShell' | 'Java';
-
 export type ProjectVersion = '~1' | '~2' | '~3';
 
+export interface ISimpleAppSettingsClient {
```

Comment:
Make sure you copy the latest api file after your recent changes, but otherwise looks good